### PR TITLE
Use healthcheck name for wrapping activity

### PR DESCRIPTION
**Pull request by AndreyTretyak**
We are currently using the constant name for the activity that wraps the health check, with this change it would be named the same as healthcheck.

### CODE REVIEW

File: src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ protected AbstractHealthCheck(TParameters parameters, ILogger logger, ActivitySo
 		/// <inheritdoc />
 		public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken token = default)
 		{
-			using Activity? activity = m_activitySource.StartActivity("HealthCheckActivity")
+			using Activity? activity = m_activitySource.StartActivity(context.Registration.Name)
```
Comment (K-Cully) [inline src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs:46]: Suggest adding an IsNullOrWhiteSpace check on this value, falling back to "HealthCheckActivity" if the name is not set correctly.

Comment (K-Cully) [inline src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs:46]: Suggest adding an IsNullOrWhiteSpace check on this value, falling back to "HealthCheckActivity" if the name is not set correctly.

Comment (AndreyTretyak) [reply src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs]: Good point added the check.

File: src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs [hunk 2]
```diff
@@ -43,7 +43,8 @@ protected AbstractHealthCheck(TParameters parameters, ILogger logger, ActivitySo
 		/// <inheritdoc />
 		public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken token = default)
 		{
-			using Activity? activity = m_activitySource.StartActivity("HealthCheckActivity")
+			string activityName = string.IsNullOrWhiteSpace(context.Registration.Name) ? "HealthCheckActivity" : context.Registration.Name;
```
Comment (artempushkin) [inline src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs:46]: Can we differentiate between activities with identical names from different services? It's very likely that there will be collisions on registration name.

Comment (AndreyTretyak) [reply src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs:46]: Activity reports service name as one of the parameters, so it should be fine. https://github.com/microsoft/Omex/blob/4bc5d95363797390b29fb8e024b6952715f4cc7c/src/Extensions/Activities/Internal/EventSource/ActivityEventSender.cs#L84

**Review by artempushkin — APPROVED** [review]

**Review by K-Cully — APPROVED** [review]

**Pull request merged**